src/platform/plugins/shared/workflows_execution_engine/server/step/http_step/http_step_impl.ts

```diff
@@ -51,56 +51,16 @@ export class HttpStepImpl extends BaseAtomicNodeImplementation<HttpStep> {
   }
 
   public getInput() {
-    const context = this.stepExecutionRuntime.contextManager.getContext();
     const { url, method = 'GET', headers = {}, body } = this.step.with;
 
     return {
-      url: typeof url === 'string' ? this.templatingEngine.render(url, context) : url,
+      url: this.stepExecutionRuntime.contextManager.renderValueAccordingToContext(url),
       method,
-      headers: this.renderHeaders(headers, context),
-      body: this.renderBody(body, context),
+      headers: this.stepExecutionRuntime.contextManager.renderValueAccordingToContext(headers),
+      body: this.stepExecutionRuntime.contextManager.renderValueAccordingToContext(body),
```

Comment:
nit - 

```suggestion
return this.stepExecutionRuntime.contextManager.renderValueAccordingToContext({'url': url, 'headers': headers, 'body': body});
```

Reply:
applied, but had to refactor all the tests due to this change 🥲